test/Interop/Cxx/operators/Inputs/member-inline.h

```diff
@@ -147,4 +147,53 @@ struct NonReferenceReadIntArray {
   }
 };
 
+struct ReadOnlyIntArrayByVal {
+  ReadOnlyIntArrayByVal(int first) { values[0] = first; }
+  ReadOnlyIntArrayByVal(const ReadOnlyIntArray &other) {}
+  int operator[](int x) const { return values[x]; }
+
+  // For testing purposes.
+  void setValueAtIndex(int value, unsigned i) { values[i] = value; }
+
+private:
+  int values[5] = { 1, 2, 3, 4, 5 };
+};
+
+struct DifferentTypesArrayByVal {
+  int operator[](int x) { return values[x]; }
+  bool operator[](bool x) { return boolValues[x]; }
+  double operator[](double x) const { return doubleValues[x == 0.0]; }
+
+private:
+  int values[3] = { 1, 2, 3 };
+  double doubleValues[2] = { 1.5, 2.5 };
+  bool boolValues[2] = { true, false };
+};
+
+template<class T> struct TemplatedArrayByVal {
+  T ts[];
+  T operator[](int i) { return ts[i]; }
+};
+typedef TemplatedArrayByVal<double> TemplatedDoubleArrayByVal;
+
+struct TemplatedSubscriptArrayByVal {
+  int *ptr;
+  template<class T> T operator[](T i) { return ptr[i]; }
+};
+
+struct NonTrivial {
+  char *Str;
+  NonTrivial() { Str = (char*)"Non-Trivial"; }
+};
+
+struct ReadOnlyNonTrivialArrayByVal {
+  NonTrivial operator[](int x) { return S; }
+private:
+  NonTrivial S;
+};
+
+// #include <iostream>
+// #include <type_traits>
+// static_assert(!std::is_trivially_default_constructible<NonTrivial>{}, "");
```

Comment:
A few things:
1. I don't think you want to construct the `is_trivially_default_constructible`. I think you want to use `is_trivially_default_constructible<>::value`.
2. I don't know why you're including iostream.
3. I'm not sure this comment is particularly helpful, maybe just add a comment "this struct is non-trivial" (or better yet, a comment saying that the type will be passed indirectly, and maybe why it will be passed indirectly).
4. It's sort of confusing when I say "non-trivial" but this isn't what I meant. The example you have is sort of interesting because C++ and Swift will treat this struct differently, but they will both still pass this type (`NonTrivial`) in registers/return it directly. It has a non-trivial constructor, but it isn't considered, by the ABI, to be non-trivial for the purposes of a call (which is the thing we care about here, because it is what determines whether this type will be passed "directly" or not, i.e., still as a pointer). However, if you make the type sufficiently large, add a custom copy constructor, or add a custom destructor then `NonTrivial` will be passed indirectly, or as an address in both Swift and C++. I think that's what we want to test here.

Reply:
@zoecarver Do you happen to know how sufficiently bit the type has to be? Bigger than the biggest register on the platform? Should I just experiment here?

Reply:
I think the best solution is to just make the destructor non-trivial, so just add `~NonTrivial() { }`.

Reply:
Great. I am actually trying both.